app/team.go

```diff
@@ -543,11 +556,11 @@ func (a *App) JoinUserToTeam(team *model.Team, user *model.User, userRequestorId
 }
 
 func (a *App) GetTeam(teamId string) (*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().Get(teamId)
-	if result.Err != nil {
-		return nil, result.Err
+	team, err := a.Srv.Store.Team().Get(teamId)
+	if err != nil {
+		return nil, err
 	}
-	return result.Data.(*model.Team), nil
+	return team, nil
```

Comment:
nit: Probably here you can just replace it with `return a.Srv.Store.Team().Get(teamId)`